NSA-3: Fix the setting of nbResults value in the pagerfanta adapters

mivancic opened the pull request:
`getSlice` method in adapters overrides the pager search result with the query parameter `!performCount` and this causes pager to not work properly.

pspanja commented:
This will need an update to unit tests as well.

pspanja commented on lib/Core/Site/Pagination/Pagerfanta/ContentSearchHitAdapter.php:
```diff
@@ -93,18 +93,13 @@ public function getSlice($offset, $length)
+        $searchResult = $this->findService->findContent($query);
 
-        // Set count for further use if returned by search engine despite !performCount (Solr, ES)
-        if (!isset($this->nbResults) && isset($this->searchResult->totalCount)) {
```
I think the problem was in setting the search result to the property, if that is removed shouldn't it be possible to keep this?

mivancic replied:
I think that we need to remove setting the `facets` property from this method too. Because this method only needs to take care of getting the slice of search result. 

Pagerfanta view render is the one that is responsible for initializing the [Pagerfanta properties](https://github.com/whiteoctober/Pagerfanta/blob/master/src/Pagerfanta/View/DefaultView.php#L49) so we only need to take care of providing the [nbResults](https://github.com/whiteoctober/Pagerfanta/blob/master/src/Pagerfanta/Pagerfanta.php#L361) in our [adapter](https://github.com/netgen/ezplatform-site-api/blob/master/lib/Core/Site/Pagination/Pagerfanta/ContentSearchHitAdapter.php#L58) which is already done.

hknezevic replied:
~~@mivancic I think setting the facets was added for the convenience, so you don't have to fetch the entire results page to retrieve facets. For instance, instead of doing~~
$pager->getCurrentPageResults();
$facets = $pager->adapter->getFacets();

~~, you can simply call a count method without having to retrieve the results, and have facets provided to you.
But I am not 100% sure. @pspanja WTYT?~~

EDIT: ah, sorry, this is the getSlice() method we're discussing, ignore the previous part of comment. But still, I believe the same reasoning was behind setting facets in this method. Same as the totalCount, the facets have global value and do not depend on the current result. And facets can be retrieved only if the search adapter has already called the getSearchResults. So in this case you can do exactly as described above: retrieve the current page results and at the same time have the facets retrieved for you. Which is kind of convenient.

pspanja replied:
Yes, but but still you shouldn't have to call `$pager->getCurrentPageResults()` just to later get facets from `$pager->adapter->getFacets()`, that should work standalone.

AFAIR the problem was caching the result here in `$this->searchResult`, and this with `$query->performCount = false`. So if you accessed the result first, `nbResults` would be `null`, but if you accessed `nbResults` first, everything was in order. That's why I think it should be possible to keep setting `ngResults` and `facets` here -- since we are not setting `$this->searchResult` any longer.

hknezevic replied:
@pspanja fair point. I think we had an issue or a pull request opened earlier because of the same reason, right?

pspanja replied:
> Same as the totalCount, the facets have global value and do not depend on the current result.

Yes, exactly. It was probably a mistake to include this in the release, deprecating it at the same moment, providing new adapters. But oh well :)

pspanja replied:
@hknezevic hum, not sure, I can't find one.

mivancic committed:
Fix the setting of nbResults value in the pagerfanta adapters

getSlice method in adapters overrides the pager search result with the query parameter performCount set to false and this causes pager to not work properly

pspanja commented:
Great, thanks @mivancic 👍
